src/util/sample.cpp

```diff
@@ -53,8 +53,14 @@ CSAMPLE* SampleUtil::alloc(SINT size) {
         // boundaries.
         return static_cast<CSAMPLE*>(
                 _aligned_malloc(sizeof(CSAMPLE) * size, kAlignment));
-#elif defined(__GNUC__) && !defined(_GLIBCXX_HAVE_ALIGNED_ALLOC)
-        // On other platforms that don't support std::aligned_alloc
+#elif defined(_GLIBCXX_HAVE_ALIGNED_ALLOC)
+        std::size_t alloc_size = sizeof(CSAMPLE) * size;
+        // The size (in bytes) must be an integral multiple of kAlignment
+        std::size_t aligned_alloc_size = alloc_size + (kAlignment - alloc_size % kAlignment);
```

Comment:
Even when the size is just right, it adds one kAlignment extra.
How about this
+= (kAlignment - 1) 
/= kAlignment
*= kAlignment

Reply:
Good catch :see_no_evil: